html/semantics/document-metadata/interactions-of-styling-and-scripting/conditionally-block-rendering-on-link-media-attr.html

```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+
+<link rel=stylesheet href=stylesheet.py>
+<link rel=stylesheet media="screen and (max-width:10px)" href=stylesheet.py?stylesNotMatchingEnvironment&delay=2>
+<h1>Dominic Farolino</h1>
+<script>
+  test(() => {
+    const h1 = document.querySelector('h1');
+    const computedColor = getComputedStyle(h1).color;
+    const expectedColor = "rgb(128, 0, 128)";
+
+    assert_equals(computedColor, expectedColor);
+    assert_equals(document.styleSheets.length, 1);
+  }, "Only the style sheet loaded via a link element whose media attribute matches the environment should be loaded synchronously");
+
+  const secondStylesheetTest = async_test("Two style sheets should be registered as style sheets for the document after 3 seconds");
```

Comment:
This test will also randomly fail: it assumes that the clock of the browser and the clock of stylesheet.py at least run at the same rate (which they may not) and that stylesheet.py will get woken up expeditiously enough to service the request before the browser's timer fires (totally no guarantee of that from OS schedulers).  Again, I'm not sure what can be done about that.

But the upshot is that there is a good chance this test will end up disabled in browser automation frameworks.  :(